aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,295 @@
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=2)]
+
+In a browser with developer tools enabled, make a request to the sample application with the URL **http://localhost:5000/redirect-rule/1234/5678**. The regex matches the request path on `redirect-rule/(.*)`, and the path is replaced with **redirected/1234/5678**. The redirect URL is sent back to the client with a 302 (Found) status code. The browser makes a new request at the redirect URL, which will appear in the browser's address bar. Since no rules in the sample application match on the redirect URL, the second request receives a 200 (OK) response from the application and the body of the response shows the redirect URL. A complete roundtrip is made to the server when a URL is *redirected*.
+
+Original Request: **http://localhost:5000/redirect-rule/1234/5678**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_redirect.png)
+
+The part of the expression contained by parentheses is called a *capture group*. The dot (`.`) of the expression means *match any character*, and the asterisk (`*`) signifies to *match the preceding character zero or more times*. Therefore, the last two path segments of the URL, `1234/5678`, are captured by capture group `(.*)`. Any value you provide in the request URL after **redirect-rule/** will be captured by this single capture group.
+
+In the replacement string, captured groups are injected into the string with the dollar sign (`$`) followed by the sequence number of the capture. The first capture group value is obtained with `$1`, the second with `$2`, and they continue in sequence for the capture groups in your regex. There is only one captured group in the redirect rule regex in the sample application, so there is only one injected group in the replacement string, which is `$1`. When the rule is applied, the URL becomes **http://localhost:5000/redirected/1234/5678**.
+
+#### URL Redirect to a secure endpoint
+Use `AddRedirectToHttps()` to redirect insecure requests to the same host and path with secure HTTPS protocol (**https://**) with the flexibility to choose the status code and port. If the status code is not supplied, the middleware will default to 302 (Found). If the port is not supplied, the middleware will default to `null`, which means the protocol will change to **https://** and the client will access the resource on port 443. The example shows how to set the status code to 301 (Moved Permanently) and change the port to 5001.
+```csharp
+var options = new RewriteOptions()
+    .AddRedirectToHttps(301, 5001);
+
+app.UseRewriter(options);
+```
+Use `AddRedirectToHttpsPermanent()` to redirect insecure requests to the same host and path with secure HTTPS protocol (**https://** on port 443). The middleware will set the status code to 301 (Moved Permanently).
```

Comment:
No difference there. I went with 301 & 5001, because the default for `AddRedirectToHttps()` is 302 & 443, and I wanted to change both defaults in the sample.